File: backend/MasApi/Models/Invoice.cs
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;

namespace MasApi.Models
{
    [Index(nameof(InvoiceNumber), IsUnique = true)]
    public class Invoice: BaseModel
    {
        [Key]
        public required Guid Id { get; set; }
        public required string InvoiceNumber { get; set; }
        public required Guid OrderId { get; set; }
        public Order? Order { get; set; }
        public Guid? CompanyId { get; set; }
        public Company? Company { get; set; }
        public required DateTime IssuedAt { get; set; }
        public required DateTime PaymentDueDate { get; set; }
        public required Enums.InvoiceStatus Status { get; set; }

        public bool UpdateStatus()
        {
            switch (Status)
            {
                case Enums.InvoiceStatus.Issued:
                    if (Order != null && Order.Status == Enums.OrderStatus.Paid)
                    {
                        Status = Enums.InvoiceStatus.Paid;
                        return true;
                    }
                    else if (Order != null && Order.Status == Enums.OrderStatus.Cancelled)
                    {
                        Status = Enums.InvoiceStatus.Cancelled;
                        return true;
                    }
                    else if (DateTime.UtcNow > PaymentDueDate)
                    {
                        Status = Enums.InvoiceStatus.Overdue;
                        return true;
                    }
                    break;
                case Enums.InvoiceStatus.Paid:
                    if (Order != null && (Order.Status == Enums.OrderStatus.Cancelled || Order.Status == Enums.OrderStatus.Returned))
                    {
                        Status = Enums.InvoiceStatus.Refunded;
                        return true;
                    }
                    break;
            }
            return false;
        }
    }
}

File: frontend/src/types/dto/invoice.ts
import { CompanyDto } from "./company";
import { OrderDto2 } from "./order";

export interface InvoiceDto {
    id: string; // UUID
    invoiceNumber: string;
    orderId: string;
    companyId: string;
    issuedAt: string; // ISO datetime
    status: string;
    paymentDueDate: string; // ISO datetime
}

export interface InvoiceDetailsDto {
    id: string;
    invoiceNumber: string;
    order: OrderDto2;
    company: CompanyDto;
    issuedAt: string; // ISO date
    paymentDueDate: string; // ISO date
    status: string;
}


export interface CreateInvoicePayload {
    companyId: string;
    orderId: string;
}

export interface UpdateInvoicePayload {
    status: string;
}
